error-handling/src/main/java/com/palantir/remoting2/errors/ServiceException.java

```diff
@@ -74,18 +81,34 @@ public final int getStatus() {
         return status;
     }
 
-    /** The format of the exception message. */
+    @Override
     public final String getMessageFormat() {
-        return messageFormat;
+        return message;
+    }
+
+    @Override
+    public final List<Arg<?>> getMessageArgs() {
+        return args;
     }
 
-    /** The parameters for the exception message. */
-    public final List<Param<?>> getMessageParams() {
-        return messageParams;
+    protected static final String formatMessage(String message, Arg<?>... args) {
+        if (args.length == 0) {
+            return message;
+        }
+
+        String[] stringifiedArgs = Arrays.stream(args)
+                .map(arg -> String.format("%s=%s", arg.getName(), toJson(arg.getValue())))
+                .toArray(String[]::new);
+
+        return String.format("%s {%s}", message, String.join(", ", stringifiedArgs));
```

Comment:
I'd highly encourage writing some JMH benchmarks for this method and making sure that it is as efficient as possible because as written this will allocate a ton of garbage with streams, joiner, and `String.format`, at a time when the service is throwing exceptions, so I'm concerned about cascading failures.